src/magicbox/MagicBox.ts

```diff
@@ -191,10 +189,9 @@ export class MagicBoxInstance {
     };
   }
 
-  public showSuggestion() {
-    this.suggestionsManager.mergeSuggestions(this.getSuggestions != null ? this.getSuggestions() : [], suggestions => {
-      this.updateSuggestion(suggestions);
-    });
+  public async showSuggestion() {
+    await this.suggestionsManager.receiveSuggestions(this.getSuggestions());
+    this.updateSuggestion(this.suggestionsManager.suggestions);
```

Comment:
So do we ultimately we need to keep a local copy of the suggestions inside the magic box? I see on lines 250 and 269 that we are retrieving them from the suggestions manager?

Reply:
I'm not exactly sure what you mean by that. Do you mean it would be preferable to keep them in MagicBox instead? Or that we should avoid MagicBox interacting with `Suggestion` altogether? If you mean the latter, it may involve moving `InputManager` in `SuggestionsManager`.
I won't change anything until I understand.

Reply:
I see now on line 215 that we are no longer updating a local copy. In this case, is `updateSuggestions` still a good name for what the method is doing?

I see that we are:
1- Setting the input text to the suggestion text.
2- Calling `onsuggestions`. Not sure what this method is doing?
3- Adding an `onSelect` handler on every suggestion where the handler is not defined and the suggestion contains text.

That's not being conveyed by the name right now.

Reply:
Good point, I'll rename `updateSuggestion` to `addSelectEventHandlers` and I'll move some code out of this method.